worker/src/RTC/RateCalculator.cpp

```diff
@@ -12,46 +12,57 @@ namespace RTC
 		MS_TRACE();
 
 		// Ignore too old data. Should never happen.
-		if (nowMs < this->oldestTimeIndex)
+		if (nowMs < this->oldestTime)
 			return;
 
 		// Increase bytes.
 		this->bytes += size;
 
 		RemoveOldData(nowMs);
 
-		// Increase the index.
-		if (++this->newestTimeIndex >= this->windowItems)
-			this->newestTimeIndex = 0;
-
-		// Newest index overlaps with the oldest one, remove it.
-		if (this->newestTimeIndex == this->oldestIndex && this->oldestIndex != -1)
+		// If the elapsed time from the last used item is greater than the
+		// window granularity, increase the index.
```

Comment:
The condition below is not accurate.

Each item in the buffer must contain the data for a certain period of time. Ie:

Scenario where each item holds 10ms of time.

`windowSize = 1000`
`windowItems = 100`
`oldestTime = 2000`

|Index|0    |1       |2        |3          | n|        
|------|----|-----|------|-------|--|
| Time Range|2000-2009|2010-2019|2020-2029|2030-2039|.. |

So, instead of having `newestTime`, we should have the start time for the newest item, and check whether the incoming data should be added to the same item or not.

Reply:
It is already there, the newestTime update is performed only when a new item is added: https://github.com/vpalmisano/mediasoup/blob/optimize-rate-calculator/worker/src/RTC/RateCalculator.cpp#L28

Reply:
Right.

I'm having a hard time reading the PR due to the member naming. I propose some naming enhancement:

newestTimeIndex -> newestItemIndex (Buffer index holding the newest item)
oldestTimeIndex -> oldestItemIndex (Buffer index holding the oldest item)
newestTime -> newestItemStartTime (Start time of the newest item)
oldestTime -> oldestItemStartTime (Start time of the oldest item)
windowSize -> windowSizeMs (Window size in milliseconds)
itemSize -> itemSizeMs (Item size in milliseconds)

It would certainly help with the code readability.

What do you think?

Reply:
I agree, much better.